### PR TITLE
Bdex/57603: EVSS Migration: rated_disabilities - match app behavior response out side of successful response

### PR DESCRIPTION
**Pull request by ahay-agile6**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- adds a `Lighthouse::ServiceException` handler that handles known errors from the Lighthouse APIs for the controller's `ExceptionHandling` class to consume
- Benefits Disability Experience
- no feature toggled added added

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#57603
- [*Link to ticket created in va.gov-team repo*](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/57603)

## Testing done

- a whole ton of unit and integration tests
- turn on the feature toggle `disability_compensation_lighthouse_rated_disabilities_provider`, create a new `LighthouseRatedDisabilitiesProvider` and call `get_rated_disabilities` with a client_id and an rsa_key corresponding to it. 
- use an ICN that Lighthouse does not have in their database to get a 404 error and see that the `LighthouseRatedDisabilitiesProvider` has the error "Veteran not identifiable." in the response body

## Screenshots
image

## What areas of the site does it impact?
- rated_disabilities if the feature toggle `disability_compensation_lighthouse_rated_disabilities_provider` is toggled on

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by mchae-nava — DISMISSED** [review]
Nice job! And really appreciate the proactivity on this Aurora. Have some questions and nitpicks, but otherwise looks good

File: lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb [hunk 1]
```diff
@@ -37,14 +38,18 @@ def transform(data)
...
           related_disability_date: DateTime.now
         )
       end
     DisabilityCompensation::ApiProvider::RatedDisabilitiesResponse.new(rated_disabilities:)
   end
 
   def decision_code_transform(decision_code_text)
-    if decision_code_text&.downcase == 'Service Connected'.downcase
+    service_connected = decision_code_text&.downcase == 'Service Connected'.downcase ||
+                        decision_code_text&.downcase == '1151 Granted'.downcase
```
[the remark this answers is not in the document]
Comment (ahay-agile6) [reply lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb:50]: Lighthouse is sending back strings from their source for the "decision code", which affects the front-end as it is looking for a "code" rather than simply a string.

"1151 Granted" has the same treatment as "Service Connected".

They are looking into if there are any other values like that.

File: lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb [hunk 2]
```diff
@@ -37,14 +38,18 @@ def transform(data)
           rated_disability_id: 0,
           rating_decision_id: 0,
           rating_percentage: rated_disability['rating_percentage'],
+          # TODO: figure out if this is important
```
[the remark this answers is not in the document]
Comment (ahay-agile6) [reply lib/disability_compensation/providers/rated_disabilities/lighthouse_rated_disabilities_provider.rb:41]: There may be some work here, but we don't know yet because we haven't seen the Lighthouse API call work in higher environments yet.

On our local environment, we saw the UI behave in the expected way, but I wanted to note this here in case there is something failing, we have something to look at first.

File: config/locales/exceptions.en.yml
[the remark this answers is not in the document]
Comment (ahay-agile6) [reply config/locales/exceptions.en.yml:1]: they cannot.
the tests depend on them

File: lib/lighthouse/service_exception.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    def self.send_error(error, service_name, lighthouse_client_id, url)
+      return error unless error.response&.key?(:status)
+
+      send_error_logs(error, service_name, lighthouse_client_id, url)
+
+      error_status = error.response[:status]
+
+      errors = get_errors_from_response(error, error_status)
+
+      error_status = error.response[:status]
```
[the remark this answers is not in the document]
Comment (ahay-agile6) [reply lib/lighthouse/service_exception.rb:35]: nope, was just an oversight from refactoring quite a bit :smile:

File: config/locales/exceptions.en.yml
[the remark this answers is not in the document]
Comment (mchae-nava) [reply config/locales/exceptions.en.yml:1]: Ah right. In that case those would have to be lifted too. Sequencing deploys on the principle of Infrastructure > Implementation, or less to more specific.

Disregard if unnecessary though!

**Review by navaeddie — APPROVED** [review]

File: config/locales/exceptions.en.yml
[the remark this answers is not in the document]
Comment (ahay-agile6) [reply config/locales/exceptions.en.yml:1]: let me try again...
the implementation depends on this being in place, haha

[the remark this answers is not in the document]
Comment (mchae-nava) [reply config/locales/exceptions.en.yml:1]: Ah, I stand corrected haha

**Review by ryan-mcneil — APPROVED** [review]
Thanks for the additional reviews and all of the tests! Looks great!
